File: tests/xcalablemp/NOT-IMPLEMENTED/gmove_in_out/F/in8.f90
! testp099.f
! loop指示文とpost/wait指示文のテスト

      program main
      include 'xmp_lib.h'
      integer,parameter:: N=1000
!$xmp nodes p(*)
!$xmp template t(N)
!$xmp distribute t(cyclic) onto p
      integer a(N), aa
      integer procs
!$xmp align a(i) with t(i)
      character(len=3) result

      procs = xmp_num_nodes()
      if(xmp_node_num().eq.1) then
         a(1) = 1
      endif
!$xmp barrier
!$xmp loop on t(i)
      do i=2, N
         if(i.ne.2) then
!$xmp wait(p(mod(i-1,procs)), 1)
         endif
!$xmp gmove in
         aa = a(i-1)
         a(i) = aa+1
         if(i.ne.N) then
!$xmp post(p(mod(i+1,procs)), 1)
         endif
      enddo

      result = 'OK '
!$xmp loop on t(i)
      do i=1, N
         if(a(i) .ne. i) result = 'NG '
      enddo

      print *, xmp_node_num(), 'testp099.f ', result

      end
      

File: tests/xcalablemp/NOT-IMPLEMENTED/39/in8.c
/*testp099.c*/
/*loop指示文とpost/wait指示文のテスト*/
#include<mpi.h>
#include<xmp.h>
#include<stdio.h>
#include<stdlib.h>      
#pragma xmp nodes p(*)
static const int N=1000;
#pragma xmp template t(0:N-1)
#pragma xmp distribute t(cyclic) onto p
int a[N],aa;
int procs;
#pragma xmp align a[i] with t(i)
int i;
char *result; 
int main(void){
   
   procs = xmp_num_nodes();
   if(xmp_node_num() == 1){
      a[1] = 1;
   }
#pragma xmp barrier
#pragma xmp loop on t1(i)
   for(i=2;i<N;i++){
      if(i!=2){
#pragma xmp wait(p((i-1)%procs),1)
      }
#pragma xmp gmove in
      aa = a[i-1];
      a[i] = aa+1;
      if(i!=N){
#pragma xmp post(p((i+1)%procs),1)
      }
   }
   result = "OK";
#pragma xmp loop on t(i)
   for(i=0;i<N;i++){
      if(a[i] != i) result = "NG";
   }
   printf("%d %s %s\n",xmp_node_num(),"testp099.c",result);
   return 0;
}    
         
      
   
